PhysicsTools/PatAlgos/plugins/PATElectronProducer.cc

```diff
@@ -323,6 +336,7 @@ void PATElectronProducer::produce(edm::Event & iEvent, const edm::EventSetup & i
 	  const edm::RefToBase<reco::GsfElectron>& elecsRef = electrons->refAt(idx);
 	  Electron anElectron(elecsRef);
 	  anElectron.setPFCandidateRef( pfRef  );
+    anElectron.setIsolationPUPPI((*PUPPIIsolation_charged_hadrons)[elePtr], (*PUPPIIsolation_neutral_hadrons)[elePtr], (*PUPPIIsolation_photons)[elePtr]);
```

Comment:
are you sure about that? I thought it has to do with whether candidate is matched to the PF Candidate, so I would expect that we need it somewhere here:
https://github.com/ishvetso/cmssw/blob/1e180758f9416a797f0726c96769674d8cac1574/PhysicsTools/PatAlgos/plugins/PATElectronProducer.cc#L505